gitlab/client.py

```diff
@@ -208,7 +208,9 @@ def __setstate__(self, state: Dict[str, Any]) -> None:
         self.__dict__.update(state)
         # We only support v4 API at this time
         if self._api_version not in ("4",):
-            raise ModuleNotFoundError(name=f"gitlab.v{self._api_version}.objects")
+            raise ModuleNotFoundError(
+                name=f"gitlab.v{self._api_version}.objects"
+            )  # pragma: no cover, dead code currently
```

Comment:
It CAN be done 😊

```
$ cat pickle-test.py
#!/usr/bin/python3 -ttu

import pickle
import gitlab

gl = gitlab.Gitlab()
saved = pickle.dumps(gl)
loaded = pickle.loads(saved)

print(loaded)

gl._api_version = "3"
saved = pickle.dumps(gl)
loaded = pickle.loads(saved)

$ ./pickle-test.py
<gitlab.client.Gitlab object at 0x7f6133933a00>
Traceback (most recent call last):
  File "/home/jlvillal/sources/github-mine/python-gitlab/./pickle-test.py", line 14, in <module>
    loaded = pickle.loads(saved)
  File "/home/jlvillal/sources/github-mine/python-gitlab/gitlab/client.py", line 211, in __setstate__
    raise ModuleNotFoundError(name=f"gitlab.v{self._api_version}.objects")
ModuleNotFoundError
```

Reply:
Haha, I was thinking that if people try to use the public interface they will get hit at the init stage so I didn't bother. It was already a bit tedious, I should take a break from this probably :joy_cat: